crates/json-rpc/src/request.rs

```diff
@@ -9,20 +10,22 @@ use serde_json::value::RawValue;
 use std::{borrow::Cow, marker::PhantomData, mem::MaybeUninit};
 
 /// `RequestMeta` contains the [`Id`] and method name of a request.
-#[derive(Clone, Debug, PartialEq, Eq)]
+#[derive(Clone, Debug)]
 pub struct RequestMeta {
     /// The method name.
     pub method: Cow<'static, str>,
     /// The request ID.
     pub id: Id,
     /// Whether the request is a subscription, other than `eth_subscribe`.
     is_subscription: bool,
+    /// Optional HTTP extensions for the request.
```

Comment:
I think the docs here are a bit misleading, while this is a type from the http crate, this doesnt have anything to do with http really, can we add a note to this here and the setter+getter that this can be used to store extradata.